internal/service/devopsguru/devopsguru_test.go

```diff
@@ -45,3 +51,23 @@ func TestAccDevOpsGuru_serial(t *testing.T) {
 
 	acctest.RunSerialTests2Levels(t, testCases, 0)
 }
+
+func testAccPreCheck(ctx context.Context, t *testing.T) {
+	conn := acctest.Provider.Meta().(*conns.AWSClient).DevOpsGuruClient(ctx)
+
+	_, err := conn.DescribeAccountHealth(ctx, &devopsguru.DescribeAccountHealthInput{})
+
+	if acctest.PreCheckSkipError(err) {
+		t.Skipf("skipping acceptance testing: %s", err)
+	}
+	// In GovCloud partitions, API calls trigger an operation error
+	var netOpErr *net.OpError
+	if errors.As(err, &netOpErr) {
+		if strings.Contains(netOpErr.Error(), "no such host") {
+			t.Skipf("skipping acceptance testing: %s", err)
+		}
+	}
```

Comment:
Alternatively, we can move this down into `acctest.PreCheckSkipError`. I'm not sure if any other services observe errors within the GovCloud partition at this layer, but I assume this is something we'd always want to skip.

Reply:
1. Yes, I think this should be moved down into `acctest.PreCheckSkipError`
2. We also have a `if dnsErr, ok := errs.As[*net.DNSError](err); ok {` check in `internal/sweep/awsv2/skip.go` for missing API endpoints -- it would be nice to be consistent with the check (if possible)

Reply:
Moved this down into `acctest.PreCheckSkipError` and matched the logic used in the awsv2 sweeper skip function.

```go
	// Ignore missing API endpoints
	if errs.IsA[*net.DNSError](err) {
		return true
	}
```